Source/control.cpp

```diff
@@ -1290,7 +1290,7 @@ void DrawInfoBox(const CelOutputBuffer &out)
 					PrintMonstHistory(monster[pcursmonst].MType->mtype);
 				}
 			} else if (pcursitem == -1) {
-				strcpy(infostr, towners[pcursmonst]._tName);
+				strcpy(infostr, towners[pcursmonst]._tName.data());

```

Comment:
string views are not guaranteed to be zero-terminated (even though this one is), so you must provide the length explicitly via `strncpy`.